tests/CMakeLists.txt

```diff
@@ -5,37 +5,44 @@ cmake_minimum_required(VERSION 3.1)
 
 set (CMAKE_CXX_STANDARD 11)
 
-project(OSQP-Wrapper-Test)
+project(OsqpEigen-Test)
+
+if(NOT TARGET OsqpEigen::OsqpEigen)
+  find_package(osqp REQUIRED)
+  find_package(OsqpEigen REQUIRED)
+  find_package(Eigen3 REQUIRED)
+  include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
+endif()
 
 find_package(GTest REQUIRED)
 include_directories(${GTEST_INCLUDE_DIRS})
 
 # QPTest
 add_executable(SparseMatrixTest SparseMatrixTest.cpp)
-target_link_libraries(SparseMatrixTest OsqpEigen osqp::osqp ${GTEST_LIBRARIES} pthread)
+target_link_libraries(SparseMatrixTest OsqpEigen::OsqpEigen osqp::osqp Eigen3::Eigen ${GTEST_LIBRARIES} pthread)
```

Comment:
Given that both `osqp` and `Eigen3` headers are included in`OsqpEigen`'s  public headers, if you turn the line `target_link_libraries(${LIBRARY_TARGET_NAME} PRIVATE osqp::osqp Eigen3::Eigen)` in `target_link_libraries(${LIBRARY_TARGET_NAME} PUBLIC osqp::osqp Eigen3::Eigen)` the main CMakeLists.txt, you do not need  anymore to manually link `osqp::osqp` and  `Eigen3::Eigen`. However, I know realized the big problem of `FindEigen3.cmake` etc etc, so I understand why we do not do that.